client/galaxy/scripts/components/Workflow/WorkflowDropdown.vue

```diff
@@ -0,0 +1,161 @@
+<template>
+    <div>
+        <b-link
+            id="workflow-dropdown"
+            class="workflow-dropdown font-weight-bold"
+            data-toggle="dropdown"
+            aria-haspopup="true"
+            aria-expanded="false"
+        >
+            <span :class="icon" />
+            {{ workflow.name }}
+        </b-link>
+        <div class="dropdown-menu" aria-labelledby="workflow-dropdown">
+            <a
+                v-for="ops in operations"
+                :key="ops.label"
+                class="dropdown-item"
+                @click="onClick(ops.event)"
+                :href="ops.url"
+            >
+                {{ ops.label }}
+            </a>
+        </div>
+    </div>
+</template>
+<script>
+import { getAppRoot } from "onload/loadConfig";
+import { Services } from "./services.js";
+export default {
+    props: ["workflow"],
+    data() {
+        return {
+            ownerOperations: [
```

Comment:
It seems like it might be cleaner/clearer to just define these operations in the component above.  Then event bindings wouldn't have to go through the string/switch statement below, and logic would be a lot simpler.